### DISCUSSION TITLE
Change SPI HW pins to lower speed

### DISCUSSION THREAD

**Discussion by zygmund2000** [Bodmer/TFT_eSPI#2888] [General]
Hi,

I have question, how to set that parameter and where to add: gpio_set_drive_capability(GPIO_example, GPIO_DRIVE_CAP_0); to SPI lines?
It is very important for me, because I have 10MHz clock from ESP32 and I need to reduce noise from this lines.
I tried to investigate it myself without success :(

Thanks

**Comment by Bodmer** [comment]
Probably the best place to ask this is the ESP32 forum.

One option to slow down edges is to add a series resistor at the output pin.

**Comment by zygmund2000** [comment]
Resistor is not enough, the best option is slow down rising and falling edges internally in MCU, ESP32 forum maybe couldn't help it, because this option should be added in this code together with definition of output SPI pins, so only I need to know where is that definition.

**Reply by Bodmer** [reply]
This is not something I have tried or investigated as it is not needed by the library. I will leave this topic as "help wanted".

Maybe reach out to the ESP32 IDF github team?

**Reply by zygmund2000** [reply]
Ok I understand you, but it would be very useful for devices like radio receivers or any other staff to make lower rf noise anywhere.
I asked you because know this code much better than any other developers, maybe someone else can help me to find where is initialization of SPI pins because I gave up :(

**Comment by Bodmer** [comment]
Have you tried setting the strength on the output pins after the SPI port is initialised? That seems like a very simple thing to try.

Example:
```
  /// Code here to init SPI port e.g. tft.init(); for the TFT_eSPI library

   // Code to control stength of SCLK on pin 18
    gpio_set_drive_capability((gpio_num_t) 18, GPIO_DRIVE_CAP_0); // weak
  //gpio_set_drive_capability((gpio_num_t) 18, GPIO_DRIVE_CAP_1);
  //gpio_set_drive_capability((gpio_num_t) 18, GPIO_DRIVE_CAP_2); // default
  //gpio_set_drive_capability((gpio_num_t) 18, GPIO_DRIVE_CAP_3); // strong
```

The strength setting will just control the driver output impedance, so if there is not much capacitance load on the pin you may see not much of a change in rise/fall times.

**Reply by zygmund2000** [reply]
I have tried but I don't know where to put it in code, so my tests just do nothing, so my question was where it should be and in witch file?

**Reply by zygmund2000** [reply]
Thank it works :)
